src/handlers/workflows.rs

```diff
@@ -4,34 +4,33 @@ use tokio::sync::RwLock;
 use warp::reply::Reply;
 use crate::workflows::registry::WorkflowRegistry;
 use crate::workflows::Workflow;
-use warp::filters::BoxedFilter;
 
 type WR<T> = Arc<RwLock<T>>;
 
-pub fn register_workflow_handler<T: WorkflowRegistry + Send + Sync + 'static>(workflow_registry: WR<T>) -> BoxedFilter<(impl Reply,)>  {
+pub fn register_workflow_handler<T: WorkflowRegistry>(workflow_registry: WR<T>) -> impl Filter<Extract = impl warp::Reply, Error = warp::Rejection> + Clone  {
```

Comment:
Maybe declare a type since it becomes quite long and will be used throughout many types.

Reply:
error[E0658]: `impl Trait` in type aliases is unstable